### PR TITLE
`parse_recon_args`: allow `--output-directory` to be `None` by default

### PR DESCRIPTION
**Pull request by thomasmfish**

### CODE REVIEW

File: src/sim_recon/cli/parsing.py [hunk 1]
```diff
@@ -167,6 +167,7 @@ def parse_recon_args(
     parser.add_argument(
         "-o",
         "--output-directory",
+        default=None,
```
Comment (PeterC-DLS) [inline src/sim_recon/cli/parsing.py]: Why is this necessary? See https://docs.python.org/3/library/argparse.html#default

Comment (thomasmfish) [reply src/sim_recon/cli/parsing.py]: It isn't, I just like being explicit and consistent with the other arguments

Comment (PeterC-DLS) [reply src/sim_recon/cli/parsing.py]: Then you're missing more code changes!

File: src/sim_recon/cli/parsing.py [hunk 2]
```diff
@@ -209,7 +210,6 @@ def parse_recon_args(
         namespace,
         ("-c/--config-path", "config_path"),
         ("-d/--data", "sim_data_paths"),
-        ("-o/--output-directory", "output_directory"),
```
Comment (thomasmfish) [inline src/sim_recon/cli/parsing.py]: @PeterC-DLS This is what allows it to pass with a value of `None`

Comment (PeterC-DLS) [reply src/sim_recon/cli/parsing.py]: Why don't you just set the `required` argument?

Comment (thomasmfish) [reply src/sim_recon/cli/parsing.py]: I'm parsing twice, so having it set as required means that passing `--help` will just throw an error about missing required arguments. Also, help needs to be added last (after the first parse) otherwise it won't be able to display all the arguments.

Comment (PeterC-DLS) [reply src/sim_recon/cli/parsing.py]: I would have added all the options before parsing the arguments and wrote a helper function to combine the parsed arguments with the configured values.

File: src/sim_recon/cli/parsing.py (hunk 1 above)
Comment (PeterC-DLS) [reply src/sim_recon/cli/parsing.py]: My point is that `default` is missing from the `add_argument` calls above if you want to consistently explicit

File: src/sim_recon/cli/parsing.py (hunk 2 above)
Comment (thomasmfish) [reply src/sim_recon/cli/parsing.py]: I've updated how this is handled now in #30 but this still changes the behaviour, so I think should be merged first.

**Review by PeterC-DLS — APPROVED** [review]
To let you add #30
